docs/proposals/inclusion-of-notes.md

```diff
@@ -0,0 +1,92 @@
+
+**IMPORTANT: This design is merely a proposal of a design that could be implemented. It DOES NOT represent a commitment by `Foam` developers to implement the features outlined in this document. This document is merely a mechanism to facilitate discussion of a possible future direction for `Foam`.**
+
+- [Introduction](#introduction)
+- [New features](#new-features)
+  - [Including a note](#including-a-note)
+  - [Include a section of a note](#include-a-section-of-a-note)
+  - [Include an attribute of a file (note property or frontmatter)](#include-an-attribute-of-a-file-note-property-or-frontmatter)
+  - [Inclusion of backlinks overview](#inclusion-of-backlinks-overview)
+  - [Introducing a query language](#introducing-a-query-language)
+
+## Introduction
+
+Initial work and thought on including a note was ignited by issue [#652](https://github.com/foambubble/foam/issues/652). Requested by a user was a likewise functionality as offered in Obsidian. This was simply the ability to include a note.
+
+Whilst researching digital gardening for my own setup, I came across an in-depth overview by [Maggie Appleton](https://maggieappleton.com/roam-garden). Showing examples of her personal Roam Research I see valuable possibilites to connect more information, if we would add additional functionalities to the possibility of including a note. This proposal displays these possible functionalities and markup.
+
+## New features
+
+### Including a note
+
+The minimal functionality is the ability to fully include a note. Markup used in Obsidian for this is `![[wikilink]]`. For Foam I would suggest to follow this syntax. Benefits being:
+
+- Adds minimal amount of knowledge required as syntax is based on the syntax of creating a wikilink.
+- Makes the auto-complete work ouf-of-the-box, without any additional code and listeners required.
+
+**Important**. A risk exists that a loop of including the same notes arises. E.g. Note A includes note B which includes note A. This needs to be prevented by the implementation and made visible to the user.
+
+### Include a section of a note
+
+It could be interesting to only include a section of a note instead of the entire note. In order to do so thse user should be able to use the following syntax:
+
+`![[wikilink#section-b]]`
+
+As a result it will include the section title + section content until the next section *or* end of file.
+
+### Include an attribute of a file (note property or frontmatter)
+
+As a user I could be interested in collecting the value of any given proeprty for a note. For example, I might want to include the tags as defined in the frontmatter of note A. This should be possible via the syntax:
+
+`![[wikilink:<property>]]`
+
+The property value should be lookedup by foam defined properties, e.g. title, **or** any property defined in the frontmatter of a note.
+
+So, the example of including the tags of a note should be:
+
+`![[wikilink:tags]]`
```

Comment:
This is interesting, I don't have this case in my knowledge base, but I can see the utility.

In terms of the syntax, I personally like `:` (to be confirmed based also on compatibility with other md based note taking tool we'd want to play nicely with).

I think in this context the string:

```
![[Daily Routine:tags]]
```

Would produce a box with the title `tags property of Daily Routine` and the value as content?

Reply:
Yes, I can imagine the default being something like:

For multiple results (e.g. tags)
```
<div>
    <div> <span>[property]</span> of <span>[note title]</span><div>
   <div>
   [values] (displayed as unordered list)
</div>
</div>
```

For single results as title or count it could be:

```
<div> <span>[property]</span> of <span>[note title]</span> <span>[value]</span><div>
```

Rough version, of course, but at least offer flexibility to users to custom style the resulting field as they'd like. Giving optionality to hide elements for example.

Reply:
this is interesting and I still like the syntax, I am only wondering how we distinguish a title with the colon (`:`) char in it, vs the `:` used as a separator for the property name.
I don't like the idea of having to escape the `:` (kinda like what one would need to do for `|`, which is used for aliases).
Ideas:
- use the `|` itself, which people already expect to escape, but that overloads the semantic of the symbol and could create confusion
- use something we would likely not see in a filepath, e.g. `>>` (`![[Daily Routine >> tags]]`). If a filepath includes it, they would have to escape the sequence (`![[Daily \>\> Routine >> tags]]` would get the `tags` property of the `Daily >> Routine` note)
- something else?

wdyt?

Reply:
Perhaps we make life easier if we would change the syntax to something like:

`(![[Daily Routine]] >> tags)`

This way we won't interfere with the Wikilinks, we just make it explicit that you include a note and perform some extraction. The addition of the parentheses is to make life of RegEx-ing this a bit easier. But other combinations are also possible, of course.